enterprise/internal/a8n/service.go

```diff
@@ -315,6 +327,7 @@ func RunChangesetJob(
 		GitApplyArgs: []string{"-p0", "--unidiff-zero"},
 		Push:         true,
 	})
+	job.Branch = ref
```

Comment:
Should we add a sanity check here?

```suggestion
	if job.Branch != "" && job.Branch != ref {
	  return nil, fmt.Errorf("ref %q doesn't match ChangesetJob's branch %q", ref, job.Branch)
	}
	job.Branch = ref
```